extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayRecorder.java

```diff
@@ -56,28 +56,42 @@ public void setApplicationCallbackClasses(Map<String, Collection<Callback>> call
+            public ActiveResult get() {
+                // Flyway beans are inactive when the datasource itself is inactive.
+                var dataSourceBean = AgroalDataSourceUtil.dataSourceInstance(dataSourceName).getHandle().getBean();
+                var dataSourceActive = dataSourceBean.checkActive();
+                if (!dataSourceActive.value()) {
+                    return ActiveResult.inactive(
+                            String.format(Locale.ROOT,
+                                    "Flyway for datasource '%s' was deactivated automatically because this datasource was deactivated.",
+                                    dataSourceName),
+                            dataSourceActive);
+                }
+
+                // Note: When quarkus.flyway.active is set to false, Flyway beans are still available.
+                //       The property only controls automatic execution on startup.
+                // TODO should we change quarkus.flyway.active (see ^) to align on other extensions?
+                //   See https://github.com/quarkusio/quarkus/issues/42244.
+                //   We'd have something like quarkus.liquibase.startup.enabled controlling startup behavior,
+                //   and *if necessary* quarkus.liquibase.active controlling bean availability
+                //   (though IMO controlling that at the datasource level would be enough).
+                return ActiveResult.active();
+            }
+        };
+    }
+
     public Function<SyntheticCreationalContext<FlywayContainer>, FlywayContainer> flywayContainerFunction(String dataSourceName,
             boolean hasMigrations,
             boolean createPossible) {
         return new Function<>() {
             @Override
             public FlywayContainer apply(SyntheticCreationalContext<FlywayContainer> context) {
-                DataSource dataSource;
-                try {
-                    dataSource = context.getInjectedReference(DataSources.class).getDataSource(dataSourceName);
-                    if (dataSource instanceof UnconfiguredDataSource) {
-                        throw DataSourceUtil.dataSourceNotConfigured(dataSourceName);
-                    }
-                } catch (ConfigurationException e) {
-                    // TODO do we really want to enable retrieval of a FlywayContainer for an unconfigured/inactive datasource?
-                    //   Assigning ApplicationScoped to the FlywayContainer
-                    //   and throwing UnsatisfiedResolutionException on bean creation (first access)
-                    //   would probably make more sense.
-                    return new UnconfiguredDataSourceFlywayContainer(dataSourceName, String.format(Locale.ROOT,
-                            "Unable to find datasource '%s' for Flyway: %s",
-                            dataSourceName, e.getMessage()), e);
-                }
-
+                // ClientProxy.unwrap is necessary for the instanceof check below.
```

Comment:
Is it still the case? Looks like the instanceof checks are gone, and I didn't notice any apparent reason looking into `flywayProducer.createFlyway(..)`.

Looking more at it ... I see that in a similar place in the liqubase extension, this unwrapping is gone